packages/cdktf/lib/terraform-asset.ts

```diff
@@ -41,16 +46,27 @@ export class TerraformAsset extends Resource {
   constructor(scope: Construct, id: string, config: TerraformAssetConfig) {
     super(scope, id);
 
-    if (!path.isAbsolute(config.path)) {
-      throw new Error(
-        `TerraformAsset path needs to be absolute, got relative path: '${config.path}'`
-      );
+    if (path.isAbsolute(config.path)) {
+      this.sourcePath = config.path;
+    } else {
+      const cdktfJsonPath = findFileAboveCwd("cdktf.json");
+      if (cdktfJsonPath) {
+        // Relative paths are always considered to be relative to cdktf.json, but operations are performed relative to process.cwd
```

Comment:
Could you add this to the docs for TerraformAsset?

Reply:
I added to `assets.mdx`. I can also update `terraform-assets.md`, but it sounded like those docs would be going away.